test/unit/autocomplete.ts

```diff
@@ -3,25 +3,35 @@ import * as sinon from 'sinon';
 import Autocomplete from '../../src/autocomplete';
 import suite from './_suite';
 
-const CATEOGORY = 'brand';
-const CATEOGORY_VALUES = ['a', 'b', 'c'];
+const CATEGORY = 'brand';
+const CATEGORY_VALUES = ['a', 'b', 'c'];
 const SUGGESTIONS = ['d', 'e', 'f'];
 const NAVIGATIONS = ['g', 'h', 'i'];
-const STATE = { h: 'j' };
 
 suite('Autocomplete', ({ expect, spy, stub }) => {
   let autocomplete: Autocomplete;
-  let autocompleteSuggestionsSelector: sinon.SinonStub;
-  let autocompleteCategoryFieldSelector: sinon.SinonStub;
-  let autocompleteCategoryValuesSelector: sinon.SinonStub;
-  let autocompleteNavigationsSelector: sinon.SinonStub;
+  let select: sinon.SinonSpy;
 
   beforeEach(() => {
-    Autocomplete.prototype.flux = <any>{ store: { getState: () => STATE } };
-    autocompleteSuggestionsSelector = stub(Selectors, 'autocompleteSuggestions').returns(SUGGESTIONS);
-    autocompleteCategoryFieldSelector = stub(Selectors, 'autocompleteCategoryField').returns(CATEOGORY);
-    autocompleteCategoryValuesSelector = stub(Selectors, 'autocompleteCategoryValues').returns(CATEOGORY_VALUES);
-    autocompleteNavigationsSelector = stub(Selectors, 'autocompleteNavigations').returns(NAVIGATIONS);
+    Autocomplete.prototype.flux = <any>{};
+    // autocompleteSuggestionsSelector = stub(Selectors, 'autocompleteSuggestions').returns(SUGGESTIONS);
+    // autocompleteCategoryFieldSelector = stub(Selectors, 'autocompleteCategoryField').returns(CATEGORY);
+    // autocompleteCategoryValuesSelector = stub(Selectors, 'autocompleteCategoryValues').returns(CATEGORY_VALUES);
+    // autocompleteNavigationsSelector = stub(Selectors, 'autocompleteNavigations').returns(NAVIGATIONS);
+    Autocomplete.prototype.select = spy((selector) => {
+      if (selector === Selectors.autocompleteSuggestions) {
+        return SUGGESTIONS;
+      }
+      if (selector === Selectors.autocompleteCategoryField) {
+        return CATEGORY;
+      }
+      if (selector === Selectors.autocompleteCategoryValues) {
+        return CATEGORY_VALUES;
+      }
+      if (selector === Selectors.autocompleteNavigations) {
+        return NAVIGATIONS;
+      }
+    });
     autocomplete = new Autocomplete();
   });
   afterEach(() => delete Autocomplete.prototype.flux);
```

Comment:
we should remove the `select` property from the prototype afterwards so that it can't accidentally cascade to other tests